backend/src/server_process_helper.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+import asyncio
+import os
+import subprocess
+import sys
+import threading
+
+import aiohttp
+from sanic import HTTPResponse, Request
+from sanic.log import logger
+
+from api import Package
+
+
+class ExecutorServerProcess:
+    def __init__(self, port: int, flags: list[str] | None = None):
+        self.process = None
+        self.stop_event = threading.Event()
+        self.finished_starting = False
+
+        self.port = port
+        self.flags = flags or []
+
+    def start_process(self):
+        server_file = os.path.join(os.path.dirname(__file__), "server.py")
+        python_location = sys.executable
+        self.process = subprocess.Popen(
+            [python_location, server_file, str(self.port), *self.flags],
+            shell=False,
+            stdin=None,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        # Create a separate thread to read and print the output of the subprocess
+        threading.Thread(
+            target=self._read_output, daemon=True, name="output reader"
+        ).start()
+
+    def stop_process(self):
+        if self.process:
+            self.stop_event.set()
+            self.process.terminate()
+            self.process.kill()
+
+    def _read_output(self):
+        if self.process is None or self.process.stdout is None:
+            return
+        for line in self.process.stdout:
+            if self.stop_event.is_set():
+                break
+            if not self.finished_starting:
+                if "Starting worker" in line.decode():
+                    self.finished_starting = True
+            print(line.decode().strip())
```

Comment:
Why `print` and not the logger?

Reply:
Because then we'd get double log prefixes (I think. I never actually tried it). So like `(time) [info]: (time) [info]: log`